data/model-db/generate_historical_data.R

```diff
@@ -0,0 +1,778 @@
+gcs_get_object(object_name = "standard-format-data/standard_adult_passage_estimate.csv",
+               bucket = gcs_get_global_bucket(),
+               saveToDisk = "data/standard-format-data/standard_adult_passage_estimate.csv",
+               overwrite = TRUE)
+passage_estimates_raw <- read_csv("data/standard-format-data/standard_adult_passage_estimate.csv")
+
+passage_estimate <- passage_estimates_raw |> 
+  # survey_location_id
+  mutate(reach = NA) |> 
+  left_join(survey_location, by = c("stream", "reach")) |>
+  select(-c(stream, reach, description)) |>
+  rename(survey_location_id = id) |>
+  # run_id
+  left_join(run, by = c("run" = "definition")) |> 
+  rename(run_id = id) |> 
+  select(-run, -description)   
+  
+try(if(any((unique(passage_estimate$survey_location_id) %in% survey_location$id) == F)) 
+  stop("Missing Survey Location ID! Please fix!"))
+try(if(any((unique(passage_estimate$run_id) %in% run$id) == F))
+  stop("Missing Run ID! Please fix!"))
+try(if(any(is.na(passage_estimate$year)))
+  stop("Missing Year! Please fix!"))
+
+gcs_upload(passage_estimate,
+           object_function = f,
+           type = "csv",
+           name = "model-db/passage_estimate.csv",
+           predefinedAcl = "bucketLevel")
+
+# daily_holding -----------------------------------------------------------------
+# TODO check run and adipose clipped
+
+gcs_get_object(object_name = "jpe-model-data/holding.csv",
+               bucket = gcs_get_global_bucket(),
+               saveToDisk = "data/model-data/holding.csv",
+               overwrite = TRUE)
+holding_raw <- read_csv("data/model-data/holding.csv")
+
+daily_holding <- holding_raw |> 
+  select(date, reach, count, latitude, longitude, stream) |> 
+  # remove all that are missing date
+  filter(!is.na(date)) |> 
+  mutate(run = "spring",
+         adipose_clipped = F) |> 
+  left_join(survey_location, by = c("stream", "reach")) |>
+  select(-c(stream, reach, description)) |>
+  rename(survey_location_id = id) |> 
+  # run_id
+  left_join(run, by = c("run" = "definition")) |> 
+  rename(run_id = id) |> 
+  select(-run, -description)  
+
+try(if(any((unique(daily_holding$survey_location_id) %in% survey_location$id) == F)) 
+  stop("Missing Survey Location ID! Please fix!"))
+try(if(any((unique(daily_holding$run_id) %in% run$id) == F))
+  stop("Missing Run ID! Please fix!"))
+try(if(any(is.na(daily_holding$date)))
+  stop("Missing Date! Please fix!"))
+
+holding_reach <- holding_raw |> 
+  group_by(stream) |> 
+  distinct(reach)
+
+gcs_upload(daily_holding,
+           object_function = f,
+           type = "csv",
+           name = "model-db/daily_holding.csv",
+           predefinedAcl = "bucketLevel")
+# annual_holding -----------------------------------------------------------------
+# TODO check run and adipose clipped
+annual_holding <- holding_raw |> 
+  mutate(year = ifelse(is.na(year), year(date), year)) |> 
```

Comment:
Looks like there are many records where there are multiple holding counts for a given reach and day. It looks like these correspond to different river miles within reaches but might be good to add an issue to double check that we are not double counting and holding data

Reply:
Thanks for flagging. I added to the open PR Liz has for holding data counts to do another check.